chef/cookbooks/ceph/providers/client.rb

```diff
@@ -60,11 +60,17 @@ def get_caps(ceph_conf, admin_keyring, keyname)
 
 def auth_set_key(ceph_conf, admin_keyring, keyname, caps)
   # try to add the key
-  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} auth get-or-create #{keyname} #{caps}"
+  caps_tmp = caps.map { |k, v| "#{k} '#{v}'" }.join(" ")
+  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} auth get-or-create #{keyname} #{caps_tmp}"
   get_or_create = Mixlib::ShellOut.new(cmd)
   get_or_create.run_command
-  if get_or_create.stderr.scan(/EINVAL.*but cap.*does not match/)
+  unless get_or_create.stderr.scan(/EINVAL.*but cap.*does not match/).empty?
     Chef::Log.info("Deleting old key with incorrect caps")
+    # don't remove custom existing caps for osd pools
+    caps_osd  = caps["osd"] + "," + get_caps(ceph_conf, admin_keyring, keyname)["osd"]
+    caps["osd"] = caps_osd.split(",").collect { |x| x.strip }.uniq.join(",")
```

Comment:
Style/SymbolProc: Pass `&:strip` as an argument to `collect` instead of a block.